q2_metadata/normalization/_norm_check_rules.py

```diff
@@ -0,0 +1,311 @@
+    if isinstance(rule_value, dict):
+        # get the current keys
+        rule_value_set = set(rule_value)
+        # get the accepted keys from the current keys
+        accepted_set = rule_value_set & set(possible_validations)
+        # if there is not more keys than the accepted ones
+        if not len(rule_value_set ^ accepted_set):
+            # get the possible conditions
+            possible_validations_2 = possible_validations['force_to_blank_if']
+            # for each of the condition/replacements for the accepted keys
+            for rule_2, rule_value_2 in rule_value['force_to_blank_if'].items():
+                # check that the content of replacements are lists
+                if isinstance(rule_value_2, list):
+                    # get the accepted conditions from the current conditions
+                    if rule_2 in possible_validations_2:
+                        return True
+    return False
+
+
+def check_normalization(rule_value):
+    """Check that the user-defined value for the
+    rule "normalization" is correctly formatted.
+
+    normalization must be a dict
+        Apply the rules of normalizing the variables values, by replacing
+        the values outside the range by a given replacement value.
+                    normalization:
+                        gated_value:
+                            - Out of bounds
+                        maximum:
+                            - int
+                            - float
+                        minimum:
+                            - int
+                            - float
+
+    Parameters
+    ----------
+    rule_value
+        Parsed rule value for the rule "normalization".
+
+    Returns
+    -------
+    bool
+        True if successful, False otherwise.
+    """
+    if isinstance(rule_value, dict):
+        possible_normalizations = {'maximum', 'minimum', 'gated_value'}
+        rule_value_set = set(rule_value)
+        if not len(rule_value_set ^ (rule_value_set & possible_normalizations)):
+            return True
+    return False
+
+
+def check_format(rule_value):
+    """Check that the user-defined value for
+    rule "format" is correctly formatted.
+
+    format must be a str
+        Apply rule of checking that the passed column in the correct format:
+        [THIS COMMAND WOULD EITHER INFER THE DTYPE (as of now) OR USE THE
+        RESULT OF AN INITIAL DTYPE GETTER FUNCTION (see _dtypes.py)]
+                    format:
+                      - bool
+                      - float
+                      - int
+                      - str
+
+    Parameters
+    ----------
+    rule_value
+        Parsed rule value for rule "format".
+
+    Returns
+    -------
+    bool
+        True if successful, False otherwise.
+    """
+    allowed_values = {'bool', 'float', 'int', 'str'}
+    if isinstance(rule_value, str) and rule_value.lower() in allowed_values:
+        return True
+    return False
+
+
+def check_blank_missing(rule_value):
```

Comment:
This check and `check_format` are pretty much the same, this could likely be put in a single function or if you would prefer to have two variables with the types set and the missing value set that would also work and avoid these two functions.

Reply:
yes, that was originally the case - I will roll back then